statescript/executor.go

```diff
@@ -206,14 +327,40 @@ func (l Launcher) ExecuteAll(state, action string, ignoreError bool) error {
 			}
 		}
 
-		if ret := execute(filepath.Join(dir, s.Name()), timeout); ret != 0 {
-			// In case of error scripts all should be executed.
-			if ignoreError {
-				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
-			} else {
-				return errors.Errorf("statescript: error executing '%s': %d",
-					s.Name(), ret)
+		retryLaterFlag := false
+
+		for { // retry-later
+			execTime := time.Now()
+
+			ret := execute(filepath.Join(dir, s.Name()), timeout)
+
+			if ret != 0 {
+
+				// In case of error scripts all should be executed.
+				if ignoreError {
+					log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
+				} else if ret == exitRetryLater {
+					err = l.handleRetryLaterError(db, s.Name(), state, execTime)
```

Comment:
I think `handleRetryLaterError` is misleading here. It is not an error in fact, so I suggest renaming to `handleRetryLater`.